study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/TestBostonAddNoSymptionsOption.java

```diff
@@ -0,0 +1,105 @@
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        this.cfgPath = cfgPath;
+        this.cfg = studyCfg;
+        this.varsCfg = varsCfg;
+
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        this.dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+    }
+
+    @Override
+    public void run(Handle handle) {
+        String studyGuid = "testboston";
+        String activityCode = "ADHOC_SYMPTOM";
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid("testboston");
+
+        ActivityDao activityDao = handle.attach(ActivityDao.class);
+
+        ActivityVersionDto currentActivityVerDto = activityDao.getJdbiActivityVersion().findByActivityCodeAndVersionTag(
+                studyDto.getId(), activityCode, "v1").get();
+
+
+        // find the adhoc symptom activity
+        ActivityDto activityDto = handle.attach(JdbiActivity.class)
+                .findActivityByStudyGuidAndCode(studyGuid, activityCode)
+                .orElseThrow(() -> new DDPException("Could not find id for activity " + activityCode + " and study id " + studyGuid));
+
+        ActivityDef currActivityDef = activityDao.findDefByDtoAndVersion(activityDto, currentActivityVerDto);
+        FormActivityDef currFormActivityDef = (FormActivityDef)currActivityDef;
+
+        PicklistQuestionDao plQuestionDao = handle.attach(PicklistQuestionDao.class);
+        JdbiQuestion jdbiQuestion = handle.attach(JdbiQuestion.class);
+
+        // find the symptoms question
+        QuestionDto symptomsQuestionDto = jdbiQuestion.findLatestDtoByStudyIdAndQuestionStableId(studyDto.getId(), "ADHOC_SYMPTOMS").get();
+
+        List<FormSectionDef> bodySections = currFormActivityDef.getSections();
+        FormSectionDef plQuestionSec = bodySections.get(0);
+        FormBlockDef formBlockDef = plQuestionSec.getBlocks().get(0);
+
+        // load the new option config
+        Config option = dataCfg.getConfig("option");
+
+        var newOption = new Gson().fromJson(ConfigUtil.toJson(option), PicklistOptionDef.class);
+
+        ActivityVersionDto versionDto = handle.attach(JdbiActivityVersion.class)
+                .getActiveVersion(currFormActivityDef.getActivityId())
+                .orElseThrow(() -> new DDPException("Could not find latest version for activity " + activityCode));
+
+        long revisionId = versionDto.getRevId();
+
+         formBlockDef.getQuestions().forEach(questionDef -> {
+            // add the option to the question
+            if (questionDef.getStableId().equalsIgnoreCase(symptomsQuestionDto.getStableId())) {
+                PicklistQuestionDef plQuestionDef = (PicklistQuestionDef) questionDef;
+                plQuestionDao.insertOption(questionDef.getQuestionId(),newOption,plQuestionDef.getAllPicklistOptions().size() + 1,revisionId);
```

Comment:
default display order gap between each option is 10. 
If we want to have new option at the end , we might need to do something like: (plQuestionDef.getAllPicklistOptions().size() * 10) + 10

Reply:
we will know in local testing but insertOption might not insert all language translations. 
May be we can add option snippet with translations to snippetts/question-adhoc-symptoms.conf and run UpdateTemplatesInPlace task !
I will let yufeng look at this because he did quite  a bit language translations lately.

Reply:
I second Sampath's suggestion for fixing display order.